packages/@aws-cdk/assertions/lib/private/messages.ts

```diff
@@ -32,6 +32,17 @@ export function hasMessage(messages: Messages, constructPath: string, props: any
   ].join('\n');
 }
 
+export function hasNoMessage(messages: Messages, constructPath: string, props: any): string | void {
+  const section: { [key: string]: SynthesisMessage } = messages;
+  const result = matchSection(filterPath(section, constructPath), props);
+
+  if (!result.match) {
+    return;
+  }
+
+  return `Stack has ${Object.keys(result.matches).length} messages.`;
```

Comment:
```suggestion
  return `Expected no matches, but stack has ${Object.keys(result.matches).length} messages.`;
```

Reply:
In addition to the above change, I think we should also be returning the matched messages in the error message.

It's possible someone looks for `annotations.hasNoError('/Default/Fred', Match.anyValue())` and they should be able to know what error message they are getting.